Here we will look at some Data fropm AirB&B for Dublin.<br>
We will look at location and clealiness ratings for some properties<br>
Then we will see how much the dirtiest listings cost!

### Setup
import libraries and airB&B_Listings data set from Data folder

In [1]:
import numpy as np
import pandas as pd

airBnBListings = pd.read_csv("Data/airBnB_Listings.csv")
airBnBListings

id                            listing_url       scrape_id  \
0         14842     https://www.airbnb.com/rooms/14842  20180819190710   
1         43908     https://www.airbnb.com/rooms/43908  20180819190710   
2         44077     https://www.airbnb.com/rooms/44077  20180819190710   
3         46517     https://www.airbnb.com/rooms/46517  20180819190710   
4         68483     https://www.airbnb.com/rooms/68483  20180819190710   
5         71813     https://www.airbnb.com/rooms/71813  20180819190710   
6         78443     https://www.airbnb.com/rooms/78443  20180819190710   
7         85148     https://www.airbnb.com/rooms/85148  20180819190710   
8         85156     https://www.airbnb.com/rooms/85156  20180819190710   
9         97259     https://www.airbnb.com/rooms/97259  20180819190710   
10       121030    https://www.airbnb.com/rooms/121030  20180819190710   
11       159889    https://www.airbnb.com/rooms/159889  20180819190710   
12       162809    https://www.airbnb.com/rooms/162809  20180819190710   
13       163398    https://www.airbnb.com/rooms/163398  20180819190710   
14       165828    https://www.airbnb.com/rooms/165828  20180819190710   
15       165829    https://www.airbnb.com/rooms/165829  20180819190710   
16       178860    https://www.airbnb.com/rooms/178860  20180819190710   
17       181823    https://www.airbnb.com/rooms/181823  20180819190710   
18       200979    https://www.airbnb.com/rooms/200979  20180819190710   
19       227966    https://www.airbnb.com/rooms/227966  20180819190710   
20       237465    https://www.airbnb.com/rooms/237465  20180819190710   
21       245893    https://www.airbnb.com/rooms/245893  20180819190710   
22       245903    https://www.airbnb.com/rooms/245903  20180819190710   
23       246560    https://www.airbnb.com/rooms/246560  20180819190710   
24       254941    https://www.airbnb.com/rooms/254941  20180819190710   
25       261248    https://www.airbnb.com/rooms/261248  20180819190710   
26       266856    https://www.airbnb.com/rooms/266856  20180819190710   
27       266935    https://www.airbnb.com/rooms/266935  20180819190710   
28       268984    https://www.airbnb.com/rooms/268984  20180819190710   
29       272650    https://www.airbnb.com/rooms/272650  20180819190710   
...         ...                                    ...             ...   
9990   27834222  https://www.airbnb.com/rooms/27834222  20180819190710   
9991   27834375  https://www.airbnb.com/rooms/27834375  20180819190710   
9992   27840728  https://www.airbnb.com/rooms/27840728  20180819190710   
9993   27840979  https://www.airbnb.com/rooms/27840979  20180819190710   
9994   27842035  https://www.airbnb.com/rooms/27842035  20180819190710   
9995   27842164  https://www.airbnb.com/rooms/27842164  20180819190710   
9996   27843188  https://www.airbnb.com/rooms/27843188  20180819190710   
9997   27845062  https://www.airbnb.com/rooms/27845062  20180819190710   
9998   27846077  https://www.airbnb.com/rooms/27846077  20180819190710   
9999   27846740  https://www.airbnb.com/rooms/27846740  20180819190710   
10000  27846985  https://www.airbnb.com/rooms/27846985  20180819190710   
10001  27847998  https://www.airbnb.com/rooms/27847998  20180819190710   
10002  27848134  https://www.airbnb.com/rooms/27848134  20180819190710   
10003  27849776  https://www.airbnb.com/rooms/27849776  20180819190710   
10004  27850058  https://www.airbnb.com/rooms/27850058  20180819190710   
10005  27850536  https://www.airbnb.com/rooms/27850536  20180819190710   
10006  27850676  https://www.airbnb.com/rooms/27850676  20180819190710   
10007  27853317  https://www.airbnb.com/rooms/27853317  20180819190710   
10008  27860886  https://www.airbnb.com/rooms/27860886  20180819190710   
10009  27860933  https://www.airbnb.com/rooms/27860933  20180819190710   
10010  27862841  https://www.airbnb.com/rooms/27862841  20180819190710   
10011  27862999  https://www.airbnb.com/rooms/27862999  20180819190710   
10012  27863600  

===============================================

How many entries and columns are in the data?

===============================================

In [2]:
airBnBListings.shape

(10020, 96)

===============================================

What are the titles of each column?

===============================================

In [3]:
# iterating the columns 
for col in airBnBListings.columns: 
    print(col) 

id
listing_url
scrape_id
last_scraped
name
summary
space
description
experiences_offered
neighborhood_overview
notes
transit
access
interaction
house_rules
thumbnail_url
medium_url
picture_url
xl_picture_url
host_id
host_url
host_name
host_since
host_location
host_about
host_response_time
host_response_rate
host_acceptance_rate
host_is_superhost
host_thumbnail_url
host_picture_url
host_neighbourhood
host_listings_count
host_total_listings_count
host_verifications
host_has_profile_pic
host_identity_verified
street
neighbourhood
neighbourhood_cleansed
neighbourhood_group_cleansed
city
state
zipcode
market
smart_location
country_code
country
latitude
longitude
is_location_exact
property_type
room_type
accommodates
bathrooms
bedrooms
beds
bed_type
amenities
square_feet
price
weekly_price
monthly_price
security_deposit
cleaning_fee
guests_included
extra_people
minimum_nights
maximum_nights
calendar_updated
has_availability
availability_30
availability_60
availability_90
availability_365
cal


=====================================================================

How many unique locations are there?

There are mutliple columns that could relate to location<br> 
Create a subset of these columns and investigate which one may be most useful<br> 

Two issues that we are worried about:<br> 
1) Reduce the amount of <i>NaN</i> entries as much as possible<br>
2) Have the naming convention for the location as consistent as possible

======================================================================

In [4]:
location_subset = airBnBListings.loc[:,["street","neighbourhood","neighbourhood_cleansed","neighbourhood_group_cleansed","city","state","zipcode"]]
location_subset

street  \
0                Dublin, Dublin City, Ireland   
1                     Dublin, Dublin, Ireland   
2                         Churchtown, Ireland   
3           Ballyboden, South Dublin, Ireland   
4              Dublin, County Dublin, Ireland   
5                   Dublin, Leinster, Ireland   
6               Stillorgan, Leinster, Ireland   
7                 Templeogue, Dublin, Ireland   
8                         Churchtown, Ireland   
9                     Dublin, Dublin, Ireland   
10                            Dublin, Ireland   
11                            Dublin, Ireland   
12                    Dublin, Dublin, Ireland   
13                    Dublin, Dublin, Ireland   
14                            Dublin, Ireland   
15                            Dublin, Ireland   
16                            Dublin, Ireland   
17                            Dublin, Ireland   
18                    Dalkey, Dublin, Ireland   
19                            Dublin, Ireland   
20                            Dublin, Ireland   
21              Lucan, County Dublin, Ireland   
22              Lucan, County Dublin, Ireland   
23                    Dublin, Dublin, Ireland   
24                 Blackrock, Dublin, Ireland   
25                     Dun Laoghaire, Ireland   
26                     Dun Laoghaire, Ireland   
27                     Dun Laoghaire, Ireland   
28                            Dublin, Ireland   
29                            Dublin, Ireland   
...                                       ...   
9990        Dublin 12, County Dublin, Ireland   
9991         Dublin 6, County Dublin, Ireland   
9992           Dublin, County Dublin, Ireland   
9993        Dublin 13, County Dublin, Ireland   
9994         Darndale, County Dublin, Ireland   
9995         Dublin 1, County Dublin, Ireland   
9996         Darndale, County Dublin, Ireland   
9997         Shankill, County Dublin, Ireland   
9998           Dublin, County Dublin, Ireland   
9999         Dublin 7, County Dublin, Ireland   
10000        Dublin 4, County Dublin, Ireland   
10001  Harold's Cross, County Dublin, Ireland   
10002        Dublin 8, County Dublin, Ireland   
10003        Dublin 8, County Dublin, Ireland   
10004        Dublin 1, County Dublin, Ireland   
10005       Dublin 14, County Dublin, Ireland   
10006        Dublin 1, County Dublin, Ireland   
10007        Dublin 8, County Dublin, Ireland   
10008        Terenure, County Dublin, Ireland   
10009          Dublin, County Dublin, Ireland   
10010          Dublin, County Dublin, Ireland   
10011        Dublin 1, County Dublin, Ireland   
10012   County Dublin, County Dublin, Ireland   
10013          Dublin, County Dublin, Ireland   
10014          Dublin, County Dublin, Ireland   
10015       Dublin 11, County Dublin, Ireland   
10016   County Dublin, County Dublin, Ireland   
10017        Dublin 7, County Dublin, Ireland   
10018        Dublin 7, County Dublin, Ireland   
10019          Dublin, County Dublin, Ireland   

                             neighbourhood neighbourhood_cleansed  \
0                               Portobello            Dublin City   
1                                Docklands            Dublin City   
2                               Churchtown  Dn Laoghaire-Rathdown   
3                                      NaN           South Dublin   
4                               Temple Bar            Dublin City   
5                               Temple Bar            Dublin City   
6                               Donnybrook  Dn Laoghaire-Rathdown   
7                                      NaN           South Dublin   
8                               Churchtown  Dn Laoghaire-Rathdown   
9                               Kilmainham            Dublin City   
10                              Sandymount            Dublin City   
11                                Beaumont            Dublin City   
12                               Blackrock  Dn Laoghaire-Rathdown   
13                              Ki

In [5]:
location_subset["neighbourhood"].unique()

array(['Portobello', 'Docklands', 'Churchtown', nan, 'Temple Bar',
       'Donnybrook', 'Kilmainham', 'Sandymount', 'Beaumont', 'Blackrock',
       'Grafton Street/St. Stephens Green', 'Raheny',
       "North City Central/O'Connell Street", 'Drumcondra', 'Santry',
       'Artane', 'Castleknock', 'Stoneybatter/Arbour Hill',
       'Ringsend/Irishtown', 'The Liberties', 'North Strand', 'Cabra',
       'Dublin Industrial Estate', 'Old City', 'Glasnevin', 'Smithfield',
       'Phibsborough', 'Terenure', 'Kimmage',
       'Phoenix Park and Surrounds', 'Dundrum', 'Bonnybrook/Darndale',
       'Clontarf', 'Islandbridge', 'Dartry', 'Pelletstown', 'Monkstown',
       'Georgian Quarter', 'Trinity College', 'Ballsbridge', 'East Wall',
       'Rathfarnham', 'Ballyfermot', 'Rathgar', 'Rialto', 'Crumlin',
       'Ardilea', 'Ballybough', 'Ranelagh and Rathmines', 'Marino',
       "Harold's Cross", 'Booterstown', 'Inchicore', 'Ballymun',
       'Fairview', 'Knocklyon', 'Mount Merrion', 'Riverston Abbe

In [6]:
location_subset["neighbourhood_cleansed"].unique()

array(['Dublin City', 'Dn Laoghaire-Rathdown', 'South Dublin', 'Fingal'],
      dtype=object)

In [7]:
location_subset["neighbourhood"].value_counts()

Docklands                              655
North City Central/O'Connell Street    648
The Liberties                          516
Phibsborough                           424
Old City                               377
Ranelagh and Rathmines                 373
Ballsbridge                            303
Smithfield                             292
Stoneybatter/Arbour Hill               274
Drumcondra                             247
Trinity College                        236
Portobello                             234
Grafton Street/St. Stephens Green      201
Temple Bar                             196
Kilmainham                             192
Rathgar                                186
Dundrum                                160
Georgian Quarter                       156
Santry                                 145
Blackrock                              140
Monkstown                              122
Donnybrook                             122
Castleknock                            122
Ringsend/Ir

In [8]:
location_subset["neighbourhood_group_cleansed"].unique()

array([nan])

In [9]:
location_subset["neighbourhood_cleansed"].isnull().value_counts().any()

True

In [10]:
location_subset["neighbourhood_cleansed"].isnull().values.any()

False

In [11]:
location_subset["neighbourhood_group_cleansed"].notnull().values.any()

False

=================== ANSWER ===============================================

Two issues that we are worried about:<br> 
1) Reduce the amount of <i>NaN</i> entries as much as possible<br>
2) Have the naming convention for the location as consistent as possible

Both Satisfied in the column <i>neighbourhood_group_cleansed</i>

=================== ANSWER ===============================================

===============================================================================

What <i>neighbourgoods</i> are in each <i>"neighbourhood_cleansed"</i> category and how many listings does each have?

===============================================================================

In [12]:
location_subset.groupby("neighbourhood_cleansed").neighbourhood.value_counts()

neighbourhood_cleansed  neighbourhood                      
Dn Laoghaire-Rathdown   Dundrum                                160
                        Blackrock                              140
                        Monkstown                              122
                        Mount Merrion                           42
                        Booterstown                             40
                        Ardilea                                 32
                        Churchtown                              31
                        Donnybrook                              20
                        Ballinteer                              16
                        Rathfarnham                             11
Dublin City             Docklands                              655
                        North City Central/O'Connell Street    648
                        The Liberties                          516
                        Phibsborough                           424
  

===============================================================================

How many times has a property in each neighborhood been reviewed?

===============================================================================

In [13]:
airBnBListings.groupby(["neighbourhood_cleansed","neighbourhood"]).number_of_reviews.sum()

neighbourhood_cleansed  neighbourhood           
Dn Laoghaire-Rathdown   Ardilea                       334
                        Ballinteer                    125
                        Blackrock                    2033
                        Booterstown                   245
                        Churchtown                    834
                        Donnybrook                    346
                        Dundrum                      1926
                        Monkstown                    2109
                        Mount Merrion                 570
                        Rathfarnham                    79
Dublin City             Addison Park                  312
                        Ardilea                        31
                        Artane                       3534
                        Ballsbridge                  7188
                        Ballybough                   1427
                        Ballyfermot                  1169
                       

===============================================================================


Create a subset of the data frame comprised of all the <i>"Review"</i> columns

i.e all columns that have "review" in the title

===============================================================================

In [14]:
review_subset=airBnBListings.loc[:,["number_of_reviews","first_review","last_review","review_scores_rating","review_scores_accuracy","review_scores_cleanliness","review_scores_checkin","review_scores_communication","review_scores_location","review_scores_value"]]
review_subset

number_of_reviews first_review last_review  review_scores_rating  \
0                     36   2011-07-11  2018-08-08                  95.0   
1                     91   2010-08-30  2018-08-13                  97.0   
2                    163   2011-03-20  2018-08-05                  94.0   
3                      2   2015-03-04  2015-10-25                 100.0   
4                     98   2011-04-04  2018-08-13                  99.0   
5                     84   2011-08-30  2016-06-06                  79.0   
6                      1   2012-09-04  2012-09-04                  77.0   
7                      0          NaN         NaN                   NaN   
8                    116   2011-05-24  2018-08-12                  95.0   
9                     37   2016-03-22  2018-08-13                  96.0   
10                   109   2011-06-18  2018-08-12                  94.0   
11                   229   2012-03-20  2018-08-17                  95.0   
12                   181   2015-06-01  2018-08-11                  96.0   
13                    70   2011-08-24  2018-07-05                  97.0   
14                    28   2011-10-25  2018-07-29                  90.0   
15                    37   2012-01-22  2018-05-15                  94.0   
16                   261   2012-09-05  2018-08-14                  97.0   
17                    46   2011-08-18  2016-03-19                  92.0   
18                    61   2012-05-09  2018-07-22                  93.0   
19                   173   2011-10-02  2018-08-15                  78.0   
20                   178   2011-12-22  2018-08-14                  89.0   
21                    16   2012-03-20  2016-08-29                  94.0   
22                    21   2012-09-03  2018-08-11                  99.0   
23                    65   2012-09-27  2018-08-08                  93.0   
24                   147   2011-11-29  2018-07-08                  95.0   
25                   126   2011-11-20  2018-08-16                  88.0   
26                    73   2013-03-18  2018-07-23                  87.0   
27                    33   2013-08-15  2018-08-10                  84.0   
28                    97   2012-01-22  2018-03-28                  95.0   
29                     4   2016-06-19  2017-08-06                  95.0   
...                  ...          ...         ...                   ...   
9990                   0          NaN         NaN                   NaN   
9991                   0          NaN         NaN                   NaN   
9992                   0          NaN         NaN                   NaN   
9993                   0          NaN         NaN                   NaN   
9994                   0          NaN         NaN                   NaN   
9995                   0          NaN         NaN                   NaN   
9996                   0          NaN         NaN                   NaN   
9997                   0          NaN         NaN                   NaN   
9998                   0          NaN         NaN                   NaN   
9999                   0          NaN         NaN                   NaN   
10000                  0          NaN         NaN                   NaN   
10001                  0          NaN         NaN                   NaN   
10002                  0          NaN         NaN                   NaN   
10003                  1   2018-08-19  2018-08-19                   NaN   
10004                  0          NaN         NaN                   NaN   
10005                  0          NaN         NaN                   NaN   
10006                  0          NaN         NaN                   NaN   
10007                  0          NaN         NaN                   NaN   
10008                  0          NaN         NaN                   NaN   
10009                  0          NaN         NaN                   NaN   
10010                  0          NaN         NaN                   NaN   
10011                  0        

=================================================

Create three new data frames comprising of just the <i>Review</i> Columns
<br><br>The first frame should only include listings with a review
<br>The second should only include listings with a 10 reviews or more
<br>The third frame should only include listings 250 reviews or more

<br><br>Once created look at the average, max, and min ratings for the subsets of listings 

=================================================

In [15]:
reviews1_or_more = review_subset.loc[review_subset["number_of_reviews"]>=1]
reviews10_or_more = review_subset.loc[review_subset["number_of_reviews"]>=10]
reviews250_or_more = review_subset.loc[review_subset["number_of_reviews"]>=250]

In [16]:
reviews1_or_more.describe()

number_of_reviews  review_scores_rating  review_scores_accuracy  \
count        8449.000000           8290.000000             8279.000000   
mean           35.867322             93.428106                9.630269   
std            52.002195              7.627452                0.718988   
min             1.000000             20.000000                2.000000   
25%             4.000000             91.000000                9.000000   
50%            14.000000             95.000000               10.000000   
75%            45.000000             99.000000               10.000000   
max           493.000000            100.000000               10.000000   

       review_scores_cleanliness  review_scores_checkin  \
count                8283.000000            8270.000000   
mean                    9.383919               9.734704   
std                     0.939432               0.650770   
min                     2.000000               2.000000   
25%                     9.000000              10.000000   
50%                    10.000000              10.000000   
75%                    10.000000              10.000000   
max                    10.000000              10.000000   

       review_scores_communication  review_scores_location  \
count                  8281.000000             8274.000000   
mean                      9.778408                9.399081   
std                       0.596930                0.770720   
min                       2.000000                2.000000   
25%                      10.000000                9.000000   
50%                      10.000000               10.000000   
75%                      10.000000               10.000000   
max                      10.000000               10.000000   

       review_scores_value  
count          8271.000000  
mean              9.273607  
std               0.852894  
min               2.000000  
25%               9.000000  
50%               9.000000  
75%              10.000000  
max              10.000000

In [17]:
reviews10_or_more.describe()

number_of_reviews  review_scores_rating  review_scores_accuracy  \
count        4956.000000           4956.000000             4956.000000   
mean           58.496973             93.814972                9.690678   
std            58.027898              4.936679                0.507623   
min            10.000000             60.000000                7.000000   
25%            19.000000             91.000000                9.000000   
50%            37.000000             95.000000               10.000000   
75%            77.000000             98.000000               10.000000   
max           493.000000            100.000000               10.000000   

       review_scores_cleanliness  review_scores_checkin  \
count                4956.000000            4956.000000   
mean                    9.477805               9.790355   
std                     0.668833               0.433506   
min                     5.000000               7.000000   
25%                     9.000000              10.000000   
50%                    10.000000              10.000000   
75%                    10.000000              10.000000   
max                    10.000000              10.000000   

       review_scores_communication  review_scores_location  \
count                  4956.000000             4956.000000   
mean                      9.824859                9.441283   
std                       0.396243                0.575994   
min                       7.000000                6.000000   
25%                      10.000000                9.000000   
50%                      10.000000                9.000000   
75%                      10.000000               10.000000   
max                      10.000000               10.000000   

       review_scores_value  
count          4956.000000  
mean              9.322236  
std               0.589924  
min               7.000000  
25%               9.000000  
50%               9.000000  
75%              10.000000  
max              10.000000

In [18]:
reviews250_or_more.describe()

number_of_reviews  review_scores_rating  review_scores_accuracy  \
count          79.000000             79.000000               79.000000   
mean          304.898734             93.632911                9.734177   
std            59.306158              4.397485                0.444593   
min           250.000000             81.000000                9.000000   
25%           263.000000             92.000000                9.000000   
50%           278.000000             95.000000               10.000000   
75%           322.000000             97.000000               10.000000   
max           493.000000             99.000000               10.000000   

       review_scores_cleanliness  review_scores_checkin  \
count                  79.000000              79.000000   
mean                    9.468354               9.797468   
std                     0.657022               0.404454   
min                     7.000000               9.000000   
25%                     9.000000              10.000000   
50%                    10.000000              10.000000   
75%                    10.000000              10.000000   
max                    10.000000              10.000000   

       review_scores_communication  review_scores_location  \
count                    79.000000               79.000000   
mean                      9.860759                9.544304   
std                       0.348409                0.526173   
min                       9.000000                8.000000   
25%                      10.000000                9.000000   
50%                      10.000000               10.000000   
75%                      10.000000               10.000000   
max                      10.000000               10.000000   

       review_scores_value  
count            79.000000  
mean              9.354430  
std               0.555577  
min               8.000000  
25%               9.000000  
50%               9.000000  
75%              10.000000  
max              10.000000

=================================================

<br><br>The first frame should include listings with a review
<br>The second should include listings with a 10 reviews or more
<br>The first frame should include listings 250 reviews or more

Do you notice anything unusual about the minimum score for the above groups.



Lets focus on the cleanliness scores<br>
Select all listings with a cleanliness rating of less than 4

=================================================

In [19]:
airBnBListings.loc[review_subset["review_scores_cleanliness"]<4]

id                            listing_url       scrape_id  \
565    3912830   https://www.airbnb.com/rooms/3912830  20180819190710   
1179   6739022   https://www.airbnb.com/rooms/6739022  20180819190710   
2269  11353114  https://www.airbnb.com/rooms/11353114  20180819190710   
3472  14455317  https://www.airbnb.com/rooms/14455317  20180819190710   
4628  17669421  https://www.airbnb.com/rooms/17669421  20180819190710   
5583  19632643  https://www.airbnb.com/rooms/19632643  20180819190710   
6432  21290247  https://www.airbnb.com/rooms/21290247  20180819190710   
6470  21376291  https://www.airbnb.com/rooms/21376291  20180819190710   
6606  21592200  https://www.airbnb.com/rooms/21592200  20180819190710   
6853  22117976  https://www.airbnb.com/rooms/22117976  20180819190710   
7180  22885985  https://www.airbnb.com/rooms/22885985  20180819190710   
8269  24820693  https://www.airbnb.com/rooms/24820693  20180819190710   
8676  25538601  https://www.airbnb.com/rooms/25538601  20180819190710   
8720  25684027  https://www.airbnb.com/rooms/25684027  20180819190710   
8934  26073828  https://www.airbnb.com/rooms/26073828  20180819190710   
9720  27455270  https://www.airbnb.com/rooms/27455270  20180819190710   

     last_scraped                                               name  \
565    2018-08-19                                     House for rent   
1179   2018-08-19                 Ensuite double bedroom in Dublin 1   
2269   2018-08-19                   Big double room near city centre   
3472   2018-08-19     Sunfilled apartment in Ashtown by Phoenix Park   
4628   2018-08-19                       One bed Apartment in Rathgar   
5583   2018-08-19                             19 Apartment AVAILABLE   
6432   2018-08-19                                   Room with a view   
6470   2018-08-19       Whopper Triple Room, great location for city   
6606   2018-08-19                    Quiet Flat close to city centre   
6853   2018-08-19  Newly refurbished modern townhouse in City Center   
7180   2018-08-19                      Nice double room in Blackrock   
8269   2018-08-19  Apt 102 - 2 Bedrooms, 2 Bath Apartment - Sleeps 3   
8676   2018-08-19  Double Bed in Beautiful Neighbourhood of Ranelagh   
8720   2018-08-19                        Double Bedroom, City Center   
8934   2018-08-19                                        Conway Sute   
9720   2018-08-19                          Cozy/Small Double Room CC   

                                                summary  \
565   Located in a quiet  Road  in Trimleston  Boote...   
1179  Spacious en suite bedroom in luxury apartment ...   
2269  Hello everyone! Looking to rent out a double r...   
3472  I think you´ll like this spacious 85m2 two bed...   
4628  My place is a one double bedroom apartment (yo...   
5583  We love meeting new people of all backgrounds ...   
6432  Own room with private ensuite, kitchen space a...   
6470  This spacious triple bedroom is the perfect re...   
6606  The accommodation includes a living room, kitc...   
6853  Newly refurbished apartment in a Georgian town...   
7180  Hi, I have a spacious double bedroom which wil...   
8269  Buckley Hall is a block of 33 apartments and a...   
8676  Double bedroom in beautiful suburb of Ranelagh...   
8720  Double bedroom available at Marion Square Park...   
8934                                                NaN   
9720  Hello there! Stay in this cosy/small double ro...   

                                                  space  \
565                                                 NaN   
1179  Specious double bedroom with en suite bathroom...   
2269                                                NaN   
3472  WEEKLY DISCOUNT! The apartment is spacious and...   
4628  Cosy apartment has coffee machine, microwave, ...   
5583                                                NaN   
6432                                                NaN   
6470  When you book this on airbnb, you have full ac...   
6606      

In [20]:
review_subset.loc[review_subset["review_scores_cleanliness"]<4]

number_of_reviews first_review last_review  review_scores_rating  \
565                   1   2014-11-09  2014-11-09                  20.0   
1179                  1   2015-09-20  2015-09-20                  60.0   
2269                  1   2016-03-03  2016-03-03                  60.0   
3472                  1   2016-09-04  2016-09-04                  40.0   
4628                  2   2017-10-18  2018-07-19                  20.0   
5583                  2   2017-10-31  2017-12-12                  40.0   
6432                  4   2017-10-16  2017-10-19                  20.0   
6470                  1   2017-11-09  2017-11-09                  40.0   
6606                  2   2017-11-10  2018-05-26                  40.0   
6853                  1   2018-01-01  2018-01-01                  40.0   
7180                  1   2018-03-17  2018-03-17                  20.0   
8269                  2   2018-06-30  2018-08-18                  60.0   
8676                  2   2018-07-01  2018-07-08                  70.0   
8720                  1   2018-06-24  2018-06-24                  20.0   
8934                  2   2018-07-25  2018-07-26                  20.0   
9720                  1   2018-08-12  2018-08-12                  60.0   

      review_scores_accuracy  review_scores_cleanliness  \
565                      2.0                        2.0   
1179                    10.0                        2.0   
2269                     6.0                        2.0   
3472                     4.0                        2.0   
4628                     4.0                        2.0   
5583                     8.0                        2.0   
6432                     2.0                        2.0   
6470                     6.0                        2.0   
6606                     6.0                        2.0   
6853                     6.0                        2.0   
7180                     2.0                        2.0   
8269                     7.0                        3.0   
8676                     8.0                        3.0   
8720                     2.0                        2.0   
8934                     2.0                        2.0   
9720                     8.0                        2.0   

      review_scores_checkin  review_scores_communication  \
565                    10.0                         10.0   
1179                   10.0                          8.0   
2269                    4.0                         10.0   
3472                    2.0                          2.0   
4628                    8.0                          6.0   
5583                   10.0                          8.0   
6432                    2.0                          2.0   
6470                    8.0                          8.0   
6606                   10.0                         10.0   
6853                   10.0                          8.0   
7180                    2.0                          2.0   
8269                    8.0                          8.0   
8676                    9.0                          9.0   
8720                    6.0                          6.0   
8934                    2.0                          2.0   
9720                   10.0                         10.0   

      review_scores_location  review_scores_value  
565                      6.0                  2.0  
1179                    10.0                 10.0  
2269                     8.0                  6.0  
3472                     6.0                  2.0  
4628                     6.0                  2.0  
5583                     6.0                  4.0  
6432                     2.0                  2.0  
6470                     8.0                  4.0  
6606                    10.0                  4.0  
6853                     8.0                  6.0  
7180                    10.0                  2.0  
8269                     6.0                  6.0  
8676                    10.0                  


================================================================================

Create a list of all neighbourhoods in Dublin that have cleanliness score under 4 and how many reviews are in this catgory 

================================================================================

In [21]:
low_cleanliness_by_nhood = airBnBListings.loc[review_subset["review_scores_cleanliness"]<5].neighbourhood.value_counts()

In [22]:
low_cleanliness_by_nhood

Phibsborough                           3
North City Central/O'Connell Street    3
Georgian Quarter                       2
Grafton Street/St. Stephens Green      2
North Strand                           1
Docklands                              1
Dartry                                 1
Smithfield                             1
Booterstown                            1
Pelletstown                            1
Kilmainham                             1
Harold's Cross                         1
Ballsbridge                            1
Beaumont                               1
Temple Bar                             1
Donnybrook                             1
Ranelagh and Rathmines                 1
Monkstown                              1
East Wall                              1
Rathgar                                1
Name: neighbourhood, dtype: int64

================================================================================
It is probably a better idea to find out what percentage of properties in each neighbourhood have poor cleanliness

Find out what percentage of the neighbourhoods listings have a cleanliness score under 4

================================================================================

In [23]:
total_reviews = airBnBListings.loc[airBnBListings.neighbourhood.isin(low_cleanliness_by_nhood.index)].groupby("neighbourhood").number_of_reviews.sum()

In [24]:
dirty_ratio = low_cleanliness_by_nhood/total_reviews
dirty_ratio.sort_values(ascending=False, inplace=True)
dirty_ratio

Booterstown                            0.002381
Pelletstown                            0.002375
Dartry                                 0.002309
Harold's Cross                         0.000818
East Wall                              0.000662
North Strand                           0.000496
Donnybrook                             0.000495
Beaumont                               0.000490
Monkstown                              0.000474
Georgian Quarter                       0.000369
Rathgar                                0.000242
Phibsborough                           0.000237
Grafton Street/St. Stephens Green      0.000224
Kilmainham                             0.000174
Ballsbridge                            0.000139
North City Central/O'Connell Street    0.000100
Ranelagh and Rathmines                 0.000095
Smithfield                             0.000093
Temple Bar                             0.000069
Docklands                              0.000067
dtype: float64

=================================================

import airBnB_Calendar data set from Data folder

How many entries and columns are in the data?<br>
What are the column names?<br>
What are the data types?<br>
Can we work with the price column in that format?


=================================================

In [25]:
airBnBCal = pd.read_csv("Data/airBnB_Calendar.csv")
print(airBnBCal.dtypes)
airBnBCal.price.unique()
airBnBCal.price.dtype

listing_id     int64
date          object
available     object
price         object
dtype: object


dtype('O')

=================================================

We will use the price column to see how expensive the unclean airBnB listings are

Write a function to convert the price column into a 
series of floats<br>

Use the <i>apply</i> function and your function to reformat the price column


=================================================

In [26]:
def convertDollarStringToInt(dollar_string):
    dollar_str = str(dollar_string).strip('$').replace(',', '')
    return float(dollar_str)

In [27]:
airBnBCalNums = airBnBCal.price.apply(convertDollarStringToInt)
airBnBCal.price = airBnBCalNums
airBnBCal.set_index("listing_id")

date available  price
listing_id                             
14842       2019-08-18         t  119.0
14842       2019-08-17         t  119.0
14842       2019-08-16         t  119.0
14842       2019-08-15         t  119.0
14842       2019-08-14         t  119.0
14842       2019-08-13         t  119.0
14842       2019-08-12         t  119.0
14842       2019-08-11         t  119.0
14842       2019-08-10         t  119.0
14842       2019-08-09         t  119.0
14842       2019-08-08         t  119.0
14842       2019-08-07         t  119.0
14842       2019-08-06         t  119.0
14842       2019-08-05         t  119.0
14842       2019-08-04         t  119.0
14842       2019-08-03         t  119.0
14842       2019-08-02         t  119.0
14842       2019-08-01         t  119.0
14842       2019-07-31         t  119.0
14842       2019-07-30         t  119.0
14842       2019-07-29         t  119.0
14842       2019-07-28         t  119.0
14842       2019-07-27         t  119.0
14842       2019-07-26         t  119.0
14842       2019-07-25         t  119.0
14842       2019-07-24         t  119.0
14842       2019-07-23         t  119.0
14842       2019-07-22         t  119.0
14842       2019-07-21         t  119.0
14842       2019-07-20         t  119.0
...                ...       ...    ...
27871082    2018-09-17         f    NaN
27871082    2018-09-16         f    NaN
27871082    2018-09-15         f    NaN
27871082    2018-09-14         f    NaN
27871082    2018-09-13         t   50.0
27871082    2018-09-12         t   50.0
27871082    2018-09-11         t   50.0
27871082    2018-09-10         t   50.0
27871082    2018-09-09         t   50.0
27871082    2018-09-08         t   50.0
27871082    2018-09-07         t   50.0
27871082    2018-09-06         t   50.0
27871082    2018-09-05         t   50.0
27871082    2018-09-04         t   50.0
27871082    2018-09-03         t   50.0
27871082    2018-09-02         f    NaN
27871082    2018-09-01         f    NaN
27871082    2018-08-31         t   50.0
27871082    2018-08-30         t   50.0
27871082    2018-08-29         t   50.0
27871082    2018-08-28         t   50.0
27871082    2018-08-27         t   50.0
27871082    2018-08-26         f    NaN
27871082    2018-08-25         f    NaN
27871082    2018-08-24         f    NaN
27871082    2018-08-23         f    NaN
27871082    2018-08-22         f    NaN
27871082    2018-08-21         f    NaN
27871082    2018-08-20         f    NaN
27871082    2018-08-19         f    NaN

[3657300 rows x 3 columns]

===============================================

We want to look at the price of listings.
Some prices are NaNs - deal with themaccordingly!

===============================================


In [28]:
airBnBCalNotNull = airBnBCal.loc[airBnBCal.price.notnull()]

===============================================

There are prices for each listing over multiple days.

Get a figure for each listing of the average shown for every day.<br>
Get an average of all the listings averages

===============================================

In [29]:
avg_listing_price = airBnBCalNotNull.groupby("listing_id").price.mean().round(0)
avg_listing_price.mean()

125.79185330347144

===============================================

We will now compare the average of all listings to the average for listings with "poor cleanliness"<br>

1) Sort out columns in one or both data sets (airBnBListings and airBnBCalender) so we can match ratings with prices
<br>(hint: listing_id)

2) Retrieve a list of all listing id with a cleanliness rating of less than 4

===============================================

In [30]:
print(avg_listing_price.head(5))
airBnBListings.rename(columns={"id":"listing_id"}, inplace = True)
airBnBListings.set_index("listing_id", inplace = True)
low_cleanliness_entries = airBnBListings.loc[airBnBListings["review_scores_cleanliness"]<4]
low_cleanliness_entries.head(5)
# low_cleanliness_entries.rename(columns={"id":"listing_id"}, inplace = True)
# low_cleanliness_entries.set_index("id")

listing_id
14842    120.0
43908     90.0
44077     48.0
46517     35.0
68483    102.0
Name: price, dtype: float64


listing_url       scrape_id  \
listing_id                                                          
3912830      https://www.airbnb.com/rooms/3912830  20180819190710   
6739022      https://www.airbnb.com/rooms/6739022  20180819190710   
11353114    https://www.airbnb.com/rooms/11353114  20180819190710   
14455317    https://www.airbnb.com/rooms/14455317  20180819190710   
17669421    https://www.airbnb.com/rooms/17669421  20180819190710   

           last_scraped                                            name  \
listing_id                                                                
3912830      2018-08-19                                  House for rent   
6739022      2018-08-19              Ensuite double bedroom in Dublin 1   
11353114     2018-08-19                Big double room near city centre   
14455317     2018-08-19  Sunfilled apartment in Ashtown by Phoenix Park   
17669421     2018-08-19                    One bed Apartment in Rathgar   

                                                      summary  \
listing_id                                                      
3912830     Located in a quiet  Road  in Trimleston  Boote...   
6739022     Spacious en suite bedroom in luxury apartment ...   
11353114    Hello everyone! Looking to rent out a double r...   
14455317    I think you´ll like this spacious 85m2 two bed...   
17669421    My place is a one double bedroom apartment (yo...   

                                                        space  \
listing_id                                                      
3912830                                                   NaN   
6739022     Specious double bedroom with en suite bathroom...   
11353114                                                  NaN   
14455317    WEEKLY DISCOUNT! The apartment is spacious and...   
17669421    Cosy apartment has coffee machine, microwave, ...   

                                                  description  \
listing_id                                                      
3912830     Located in a quiet  Road  in Trimleston  Boote...   
6739022     Spacious en suite bedroom in luxury apartment ...   
11353114    Hello everyone! Looking to rent out a double r...   
14455317    I think you´ll like this spacious 85m2 two bed...   
17669421    My place is a one double bedroom apartment (yo...   

           experiences_offered  \
listing_id                       
3912830                   none   
6739022                   none   
11353114                  none   
14455317                  none   
17669421                  none   

                                        neighborhood_overview  \
listing_id                                                      
3912830     Enjoying a quiet, peaceful location there are ...   
6739022     Quiet, mainly expats. Not many restaurants or ...   
11353114                                                  NaN   
14455317    This is a nice and safe residential area with ...   
17669421    This Southside area is one of the city's most ...   

                                                        notes  \
listing_id                                                      
3912830                                                   NaN   
6739022     Tenants work during the day so will only be ab...   
11353114                                                  NaN   
14455317    This is a fresh, modern and bright apartment t...   
17669421    The apartment has all information for tourists...   

                  ...        requires_license license jurisdiction_names  \
listing_id        ...                                                      
3912830           ...                       f     NaN                NaN   
6739022           ...                       f     NaN                NaN   
11353114          ...                       f     NaN                NaN   
14455317          ...                       f     NaN                NaN   
17669421          ...                       f     NaN

=======================================================================

Construct a list of (average) prices for all listings whose cleranlimess rating is less than 4 

=======================================================================

In [31]:
low_cleanliness_entries["AVG Price"]=avg_listing_price
low_cleanliness_entries.head(3)

C:\Program Files (x86)\Microsoft Visual Studio\Shared\Anaconda3_64\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


listing_url       scrape_id  \
listing_id                                                          
3912830      https://www.airbnb.com/rooms/3912830  20180819190710   
6739022      https://www.airbnb.com/rooms/6739022  20180819190710   
11353114    https://www.airbnb.com/rooms/11353114  20180819190710   

           last_scraped                                name  \
listing_id                                                    
3912830      2018-08-19                      House for rent   
6739022      2018-08-19  Ensuite double bedroom in Dublin 1   
11353114     2018-08-19    Big double room near city centre   

                                                      summary  \
listing_id                                                      
3912830     Located in a quiet  Road  in Trimleston  Boote...   
6739022     Spacious en suite bedroom in luxury apartment ...   
11353114    Hello everyone! Looking to rent out a double r...   

                                                        space  \
listing_id                                                      
3912830                                                   NaN   
6739022     Specious double bedroom with en suite bathroom...   
11353114                                                  NaN   

                                                  description  \
listing_id                                                      
3912830     Located in a quiet  Road  in Trimleston  Boote...   
6739022     Spacious en suite bedroom in luxury apartment ...   
11353114    Hello everyone! Looking to rent out a double r...   

           experiences_offered  \
listing_id                       
3912830                   none   
6739022                   none   
11353114                  none   

                                        neighborhood_overview  \
listing_id                                                      
3912830     Enjoying a quiet, peaceful location there are ...   
6739022     Quiet, mainly expats. Not many restaurants or ...   
11353114                                                  NaN   

                                                        notes    ...     \
listing_id                                                       ...      
3912830                                                   NaN    ...      
6739022     Tenants work during the day so will only be ab...    ...      
11353114                                                  NaN    ...      

           license jurisdiction_names instant_bookable  \
listing_id                                               
3912830        NaN                NaN                f   
6739022        NaN                NaN                f   
11353114       NaN                NaN                f   

           is_business_travel_ready  cancellation_policy  \
listing_id                                                 
3912830                           f             moderate   
6739022                           f             moderate   
11353114                          f             flexible   

            require_guest_profile_picture require_guest_phone_verification  \
listing_id                                                                   
3912830                                 f                                f   
6739022                                 f                                f   
11353114                                f                                f   

            calculated_host_listings_count  reviews_per_month AVG Price  
listing_id                                                               
3912830                                  1               0.02     161.0  
6739022                                  1               0.03       NaN  
11353114                                 1               0.03       NaN  

[3 rows x 96 columns]

In [32]:
# low_cleanliness_entries.head()
cleanliness_vs_price = low_cleanliness_entries.loc[:,["review_scores_cleanliness","AVG Price"]]
cleanliness_vs_price

review_scores_cleanliness  AVG Price
listing_id                                      
3912830                           2.0      161.0
6739022                           2.0        NaN
11353114                          2.0        NaN
14455317                          2.0      160.0
17669421                          2.0        NaN
19632643                          2.0        NaN
21290247                          2.0       45.0
21376291                          2.0       88.0
21592200                          2.0        NaN
22117976                          2.0        NaN
22885985                          2.0        NaN
24820693                          3.0      114.0
25538601                          3.0        NaN
25684027                          2.0        NaN
26073828                          2.0       90.0
27455270                          2.0       75.0

========================  =============================
Are properties with a poor cleanliness rating cheaper on average than all properties in Dublin?
========================  =============================

In [33]:
cleanliness_vs_price.dropna()["AVG Price"].mean()

104.71428571428571

======================== Answer =============================<br>
Population - 125.79185330347144 <br>
poor cleanliness - 104.71428571428571<br>
Cheaper but not by much!<br>

======================== Answer =============================

================================================================================
================================================================================
================================================================================
================================================================================
================================================================================
================================================================================
================================================================================
================================================================================
================================================================================

import airB&B_Reviews data set from Data folder

How many entries and columns are in the data?
What are the column names?

========================================

In [34]:
airBnBReviews = pd.read_csv("Data/airBnB_Reviews.csv")
airBnBReviews.head(5)

listing_id       id        date  reviewer_id reviewer_name  \
0       14842   366404  2011-07-11       381202        Gianni   
1       14842   436714  2011-08-11       393419        Andrea   
2       14842   825239  2012-01-02       205507       Rebecca   
3       14842  1903801  2012-08-06      1809135         Shane   
4       14842  2072172  2012-08-23       720139         Rowan   

                                            comments  
0  Hilary and her husband were very welcoming and...  
1  Great place. Great location. Hilary was extrem...  
2  Unfortunately Hilary's house cannot be recomme...  
3  I had an absolutely amazing time at Hillary's ...  
4  We stayed here for two weeks and very much enj...

In [35]:
# iterating the columns 
for col in airBnBReviews.columns: 
    print(col)

listing_id
id
date
reviewer_id
reviewer_name
comments


================================================================================



================================================================================

In [36]:
low_cleanliness_entries = airBnBListings.loc[airBnBListings["review_scores_cleanliness"]<4]
id_and_cleanliness= low_cleanliness_entries.loc[:,["id","review_scores_cleanliness"]]
id_and_cleanliness.rename(columns={"id":"listing_id"}, inplace = True)
id_and_cleanliness.set_index("listing_id")

C:\Program Files (x86)\Microsoft Visual Studio\Shared\Anaconda3_64\lib\site-packages\pandas\core\indexing.py:1472: FutureWarning: 
Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.

See the documentation here:
https://pandas.pydata.org/pandas-docs/stable/indexing.html#deprecate-loc-reindex-listlike
  return self._getitem_tuple(key)


review_scores_cleanliness
listing_id                           
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               2.0
NaN                               3.0
NaN                               3.0
NaN                               2.0
NaN                               2.0
NaN                               2.0

In [37]:
listings_and_review = low_cleanliness_entries.join(airBnBReviews)